tests/models/anthropic/test_output.py

```diff
@@ -0,0 +1,407 @@
+    agent = Agent(model)
+
+    @agent.tool_plain
+    def get_weather(location: str) -> str:  # pragma: no cover
+        return f'Weather in {location}'
+
+    agent.run_sync('What is the weather in Paris?')
+
+    completion_kwargs = get_mock_chat_completion_kwargs(mock_client)[0]
+    tools = completion_kwargs['tools']
+    betas = completion_kwargs['betas']
+    assert tools == snapshot(
+        [
+            {
+                'name': 'get_weather',
+                'description': '',
+                'input_schema': {
+                    'type': 'object',
+                    'properties': {'location': {'type': 'string'}},
+                    'additionalProperties': False,
+                    'required': ['location'],
+                },
+                'strict': True,
+            }
+        ]
+    )
+    assert betas == snapshot(['structured-outputs-2025-11-13'])
+
+
+def test_anthropic_strict_tools_sonnet_4_0(allow_model_requests: None, weather_tool_responses: list[BetaMessage]):
+    """Test that strict is NOT sent for non-supporting models."""
+    mock_client = MockAnthropic.create_mock(weather_tool_responses)
+    model = AnthropicModel('claude-sonnet-4-0', provider=AnthropicProvider(anthropic_client=mock_client))
+    agent = Agent(model)
+
+    @agent.tool_plain
+    def get_weather(location: str) -> str:  # pragma: no cover
+        return f'Weather in {location}'
+
+    agent.run_sync('What is the weather in Paris?')
+
+    completion_kwargs = get_mock_chat_completion_kwargs(mock_client)[0]
+    tools = completion_kwargs['tools']
+    betas = completion_kwargs.get('betas')
+    assert tools == snapshot(
+        [
+            {
+                'name': 'get_weather',
+                'description': '',
+                'input_schema': {
+                    'type': 'object',
+                    'properties': {'location': {'type': 'string'}},
+                    'additionalProperties': False,
+                    'required': ['location'],
+                },
+            }
+        ]
+    )
+    assert betas is OMIT
+
+
+async def test_anthropic_native_output_multiple(
+    allow_model_requests: None,
+    anthropic_model: AnthropicModelFactory,
+    city_location_schema: type[BaseModel],
+    country_language_schema: type[BaseModel],
+):
+    """Test native output with union of multiple schemas."""
+    model = anthropic_model('claude-sonnet-4-5')
+    agent = Agent(model, output_type=NativeOutput([city_location_schema, country_language_schema]))
+
+    @agent.tool_plain
+    async def get_user_country() -> str:
+        return 'France'
+
+    result = await agent.run('What is the capital of the user country?')
+    # Should return CityLocation since we asked about capital
+    assert isinstance(result.output, city_location_schema | country_language_schema)
+    if isinstance(result.output, city_location_schema):  # pragma: no branch
```

Comment:
We shouldn't need this check if we make the assert explicitly verify `city_location_schema`